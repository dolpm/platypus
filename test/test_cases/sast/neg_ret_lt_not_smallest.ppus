pipe test |> ['a, 'b] |> [x: &'a int, y: &'b int] |> &'a int {
  if (@x == 2) {
    |> x;
  } else {
    |> y;
  }
}

pipe main |> [] |> [] |> unit {
  int x <| 4;
  |> ();
}
