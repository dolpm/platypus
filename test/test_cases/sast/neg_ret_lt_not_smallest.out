Fatal error: exception Failure("lifetime of return value in test must be the smallest (rightmost) lifetime of all possible returned arguments: 'b")
